ydb/core/kafka_proxy/actors/kafka_balance_actor_sql.cpp

```diff
@@ -0,0 +1,155 @@
+#include "kafka_balancer_actor.h"
+
+namespace NKafka {
+
+// savnik add db
+
+const TString SELECT_STATE_AND_GENERATION = R"(
+    --!syntax_v1
+    DECLARE $ConsumerGroup AS Utf8;
+
+    SELECT state, generation
+    FROM kafka_connect_groups
+    WHERE consumer_group = $ConsumerGroup;
+)";
+
+const TString INSERT_NEW_GROUP = R"(
+    --!syntax_v1
+    DECLARE $ConsumerGroup AS Utf8;
+    DECLARE $Generation AS Uint64;
+    DECLARE $State AS Uint64;
+
+    INSERT INTO kafka_connect_groups
+    (
+        consumer_group,
+        generation,
+        state,
+        current_generation_start_time
+    )
+    VALUES
+    (
+        $ConsumerGroup,
+        $Generation,
+        $State,
+        CurrentUtcDateTime()
+    );
+)";
+
+const TString UPDATE_GROUP = R"(
+    --!syntax_v1
+    DECLARE $ConsumerGroup AS Utf8;
+    DECLARE $NewState AS Uint64;
+    DECLARE $OldGeneration AS Uint64;
+
+    UPDATE kafka_connect_groups
+    SET
+        state = $NewState,
+        generation = $OldGeneration + 1
+    WHERE consumer_group = $ConsumerGroup;
+)";
+
+const TString SELECT_MASTER = R"(
+    --!syntax_v1
+    DECLARE $ConsumerGroup AS Utf8;
+    DECLARE $Generation AS Uint64;
+
+    SELECT member_id
+    FROM kafka_connect_members
+    WHERE consumer_group = $ConsumerGroup
+      AND generation = $Generation
+    ORDER BY join_time
+    LIMIT 1;
+)";
+
+const TString INSERT_MEMBER_AND_SELECT_MASTER = R"(
```

Comment:
In code you using INSERT_MEMBER_AND_SELECT_MASTER_QUERY. No _QUERY suffix here.
And you should check that visibility of transaction changes is enabled. What if min join-time in this request?